state/addmachine.go

```diff
@@ -536,6 +536,31 @@ type machineStorageParams struct {
 	filesystemAttachments map[names.FilesystemTag]FilesystemAttachmentParams
 }
 
+func combineMachineStorageParams(lhs, rhs *machineStorageParams) *machineStorageParams {
+	out := &machineStorageParams{}
+	out.volumes = append(lhs.volumes[:], rhs.volumes...)
+	out.filesystems = append(lhs.filesystems[:], rhs.filesystems...)
+	if lhs.volumeAttachments != nil || rhs.volumeAttachments != nil {
+		out.volumeAttachments = make(map[names.VolumeTag]VolumeAttachmentParams)
+		for k, v := range lhs.volumeAttachments {
+			out.volumeAttachments[k] = v
+		}
+		for k, v := range rhs.volumeAttachments {
+			out.volumeAttachments[k] = v
```

Comment:
I presume it's by design that the rhs wins if the same key appears in both sides?